src/coap_tcp.c

```diff
@@ -233,7 +232,10 @@ coap_socket_bind_tcp(coap_socket_t *sock,
     coap_log(LOG_ALERT, "coap_socket_bind_tcp: unsupported sa_family\n");
   }
 
-  if (bind(sock->fd, &listen_addr->addr.sa, listen_addr->size) == COAP_SOCKET_ERROR) {
+  if (bind(sock->fd, &listen_addr->addr.sa,
+           listen_addr->addr.sa.sa_family == AF_INET ?
```

Comment:
This should not be necessary. If `local_if` is properly set, `local_if.size` would be `sizeof(struct sockaddr_in)` for family AF_INET.

Reply:
This addresses issue fixed  #396 which was fixed in coap_io.c and this code is now a part of coap_tcp.c